microsoft/testsuites/lis/lissuite.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CentOs, Redhat
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Cat, Modinfo, Wget
+from lisa.util import SkippedException, get_matched_str
+
+
+@TestSuiteMetadata(
+    area="lis",
+    category="functional",
+    description="""
+        This test suite contains tests that are dependent on an LIS driver
+    """,
+    requirement=simple_requirement(
+        supported_os=[
+            CentOs,
+            Redhat,
+        ]
+    ),
+)
+class Lis(TestSuite):  # noqa
+    @TestCaseMetadata(
+        description="""
+            Downloads header files based on the LIS version and compares the installed
+             LIS version to the expected one which is found in the header files.
+
+            Steps:
+            1. Check for RPM
+            2. Capture installed LIS version on the node
+            3. For each rhel version (5,6,7), it downloads the header file and compares
+             the LIS version in the header file with the LIS version installed
+        """,
+        priority=1,
+    )
+    def verify_lis_driver_version(self, node: Node, log: Logger) -> None:
+        cat = node.tools[Cat]
+        modinfo_tool = node.tools[Modinfo]
+        wget_tool = node.tools[Wget]
+        node.tools[LisDriver]
+
+        # Checking for RPM Package Manager
+        rpm_qa = node.execute("rpm -qa").stdout
+        if ("kmod-microsoft-hyper-v" not in rpm_qa) or (
+            "microsoft-hyper-v" not in rpm_qa
+        ):
+            raise SkippedException("No LIS RPM's are detected. Skipping test.")
+
+        # Capturing LIS version from the node
+        version = modinfo_tool.get_version(mod_name='"hv_vmbus"').strip()
+
+        for i in range(5, 8):
+            node.execute("rm -rf hv_compat.h")
+            wget_tool.get(
+                "https://raw.githubusercontent.com/LIS/lis-next/"
+                f"{version}/hv-rhel{i}.x/hv/include/linux/hv_compat.h",
+                filename="hv_compat.h",
+                file_path="./",
+            )
+
+            # Capturing the LIS version from the source code
+            source_version = cat.read_with_filter(
+                "hv_compat.h", "define HV_DRV_VERSION"
+            )
+            version_pattern = re.compile(r'"(.+)"')
```

Comment:
1. move regexp to class level, so it doesn't need to be compiled every time. It's a small perf impact, but a good practice.
2. add examples above the patterns. So if it missed some case, people can test the updated pattern with previous examples.

Reply:
Done